widget/menu_item.go

```diff
@@ -245,6 +271,24 @@ func (r *menuItemRenderer) Refresh() {
 		r.background.Hide()
 	}
 	r.background.Refresh()
+
+	if r.i.Item.Disabled {
+		r.text.Color = theme.DisabledColor()
+		// TODO: enable after fixing translucence and alpha channel (https://github.com/fyne-io/fyne/issues/1977)
```

Comment:
This is not the right way - it should be changed to a DisabledResource and the colour will update appropriately.
You could look at the existing `Button` widget that disables icons as appropriate.